HarfangLab/harfanglab/asset_connector/device_assets.py

```diff
@@ -129,18 +132,18 @@ def map_fields(self, asset: dict[str, Any]) -> DeviceOCSFModel:
             device=device,
         )
 
-    def __fetch_devices(self, from_date: datetime) -> Generator[list[dict[str, Any]], None, None]:
+    def __fetch_devices(self, from_date: str | None) -> Generator[list[dict[str, Any]], None, None]:
         devices_url = urljoin(self.base_url, self.AGENT_ENDPOINT)
-        offset = 0
-        firstseen_param: str = from_date.isoformat()
+        firstseen_param: str = from_date if from_date else ""
 
         params: dict[str, Union[str, int]] = {
             "ordering": self.DEVICE_ORDERING_FIELD,
-            "firstseen": firstseen_param,
             "limit": self.limit,
-            "offset": offset,
         }
 
+        if firstseen_param:
+            params["firstseen"] = firstseen_param
```

Comment:
Proposal
```suggestion
        if from_date:
            params["firstseen"] = from_date
```
And we can remove L137